Routines/Necromancer/NEC_Assist.cs

```diff
@@ -0,0 +1,2207 @@
+            if  (!IsValidTarget(target))
+                return false;
+
+            position = target.Position;
+            return true;
+        }
+
+        TrinityActor lastCSTarget = null;
+        double lastCSCast = 0;
+        protected override bool ShouldCommandSkeletons(out TrinityActor target)
+        {
+            target = null;
+
+            if (MyGetTickCount() - lastCSCast > 2000)
+                lastCSTarget = null;
+            
+            if (!Skills.Necromancer.CommandSkeletons.CanCast())
+                return false;
+
+            if (Skills.Necromancer.LandOfTheDead.TimeSinceUse < 10000)
+            {
+                if (Skills.Necromancer.LandOfTheDead.CanCast())
+                    return false;
+                
+                if (CastCSInterval !=0 && Skills.Necromancer.CommandSkeletons.TimeSinceUse < CastCSInterval*10)
+                    return false;
+            }
+            else 
+            {
+                int interval = 250;
+                if (Skills.Necromancer.LandOfTheDead.CanCast())
+                    interval = 500;
+                if (Skills.Necromancer.CommandSkeletons.TimeSinceUse < interval)
+                    return false;
+            }
+
+            if (false)
+                target = GetBestTarget2(Player.Position, 60f, true, lastCSTarget)??GetBestTarget2(Player.Position, 60f, true)??GetBestTarget2(Player.Position, 60f, false)??TargetUtil.GetClosestUnit(30f);
+            else 
+                target = GetBestTarget(Player.Position, 60f, true);
+
+            if (!IsValidTarget(target))
+                return false;
+
+            // Core.Logger.Warn($"使用号令骸骨刷CD，目标距离我{target.Distance}");
+
+            lastCSTarget = target;
+            lastCSCast = MyGetTickCount();
+
+            return true;
+        }  
+
+        protected override bool ShouldBoneSpikes(out TrinityActor target)
+        {
+            target = null;
+
+            if (!Skills.Necromancer.BoneSpikes.CanCast())
+                return false;
+
+            target = BestClusterUnit(Player.Position, 60f, true)??TargetUtil.GetClosestUnit(30f);
+            return IsValidTarget(target);
+        }
+
+        protected override bool ShouldSiphonBlood(out TrinityActor target)
+        {
+            target = null;
+            if (!Skills.Necromancer.SiphonBlood.CanCast())
+                return false;
+
+            target = GetBestTarget(Player.Position, 50f);
+            return IsValidTarget(target);
+        }
+
+        protected override bool ShouldCommandGolem(out TrinityActor target)
+        {
+            target = null;
+
+            // note: this is returning PowerManager.CanCast true even when active is on cooldown.
+
+            if (!Skills.Necromancer.CommandGolem.CanCast())
+                return false;
+
+            if (Skills.Necromancer.CommandGolem.TimeSinceUse < 4000)
+                return false;
+
+            var rune = Skills.Necromancer.CommandGolem.CurrentRune;
```

Comment:
using a switch would be better instead of so many if statements, and remove the empty ones.